docs/testing/cypress.mdx

```diff
@@ -0,0 +1,191 @@
+
+This guide aims to help you set up your environment for creating Clerk-authenticated tests with Cypress.
+
+> [!NOTE]
+> Check out the [demo repo](https://github.com/clerk/example-cypress-nextjs) that tests a Clerk-powered application using [Testing Tokens](/docs/testing/overview#testing-tokens).
+
+<Steps>
+### Install `@clerk/testing`
+
+Clerk's testing package provides integration helpers for popular testing frameworks. Install it by running the following command:
+
+<CodeBlockTabs options={['npm', 'yarn', 'pnpm']}>
+
+```sh {{ filename: 'terminal' }}
+npm i @clerk/testing --save-dev
+```
+
+```sh {{ filename: 'terminal' }}
+yarn add -D @clerk/testing
+```
+
+```sh {{ filename: 'terminal' }}
+pnpm add @clerk/testing -D
+```
+
+</CodeBlockTabs>
+
+
+### Set your API keys
+
+In your test runner, set your publishable and secret key as the `CLERK_PUBLISHABLE_KEY` and `CLERK_SECRET_KEY` environment variables, respectively.
+
+To find your keys:
+1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=api-keys).
+2. In the navigation sidebar, select **API Keys**.
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+
+> [!WARNING]
+> Ensure you provide the secret key in a secure manner, to avoid leaking it to third parties. For example, if you are using GitHub Actions, refer to [*Using secrets in GitHub Actions*](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions).
+
+
+### Global setup
+
+To set up Clerk with Cypress, add the following code to your [`cypress.config.ts`](https://docs.cypress.io/guides/references/configuration) file.
+
+```tsx filename="cypress.config.ts"
+import { clerkSetup } from '@clerk/testing/cypress';
+import { defineConfig } from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      return clerkSetup({ config });
+    },
+    baseUrl: 'http://localhost:3000' // your app's URL
+  }
+});
+```
+
+`clerkSetup` will obtain a [Testing Token](/docs/testing/overview#testing-tokens) when your test suite starts, making it available for all subsequent tests to use.
+
+### Use Clerk Testing Tokens
+
+Now that Cypress is configured with Clerk, you can use the `setupClerkTestingToken()` function in your tests to augment them with the Testing Token. See the following example:
+
+```tsx filename="my-test.cy.ts"
+import { setupClerkTestingToken } from "@clerk/testing/cypress";
+
+it("sign up", () => {
+  setupClerkTestingToken();
+
+  cy.visit('/sign-up');
+  // ...
+});
+```
+
+</Steps>
+
+## Custom Commands
+
+The `@clerk/testing` package also provides [Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands) to sign in/sign out with Clerk in your Cypress tests without having to interact with the UI.
+To use these commands, you must import them in your `cypress/support/commands.ts` file.
```

Comment:
they _need_ to use them in this specific file? or they _can_ use them like the example shows?

Reply:
Hmm, they need to use them in their support file, so they'll be loaded before any test is run. 
The support file path for E2E tests could be one of the following:
- cypress/support/e2e.js
- cypress/support/e2e.jsx
- cypress/support/e2e.ts
- cypress/support/e2e.tsx

It's a common practice to have a `commands.ts/js` file in the `support` folder and import it inside the `e2e.ts/js`, that's why I mentioned this file here, but we could rephrase it to be clearer. Do you have any suggestions on this? 🤔 
support file ref: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file